libraries/botframework-schema/src/index.ts

```diff
@@ -17,939 +19,1122 @@ export * from './teams';
  * Attachment View name and size
  */
 export interface AttachmentView {
-  /**
-   * Id of the attachment
-   */
-  viewId: string;
-  /**
-   * Size of the attachment
-   */
-  size: number;
+    /**
+     * Id of the attachment
+     */
+    viewId: string;
+    /**
+     * Size of the attachment
+     */
+    size: number;
 }
 
+export const assertAttachmentView: Assertion<AttachmentView> = (value, path) => {
+    assert.unsafe.objectAs<AttachmentView>(value, path);
+    assert.string(value.viewId, path.concat('viewId'));
+    assert.number(value.size, path.concat('size'));
+};
+
+const assertAttachmentViewArray: Assertion<Array<AttachmentView>> = assert.arrayOf(assertAttachmentView);
+
+export const isAttachmentView = assert.toTest(assertAttachmentView);
+
 /**
  * Metadata for an attachment
  */
 export interface AttachmentInfo {
-  /**
-   * Name of the attachment
-   */
-  name: string;
-  /**
-   * ContentType of the attachment
-   */
-  type: string;
-  /**
-   * attachment views
-   */
-  views: AttachmentView[];
-}
+    /**
+     * Name of the attachment
+     */
+    name: string;
+    /**
+     * ContentType of the attachment
+     */
+    type: string;
+    /**
+     * attachment views
+     */
+    views: AttachmentView[];
+}
+
+export const assertAttachmentInfo: Assertion<AttachmentInfo> = (value, path) => {
+    assert.unsafe.objectAs<AttachmentInfo>(value, path);
+    assert.string(value.name, path.concat('name'));
+    assert.array(value.views, path.concat('views'));
+    assertAttachmentViewArray(value.views, path.concat('views'));
+};
+
+export const isAttachmentInfo = assert.toTest(assertAttachmentInfo);
```

Comment:
It's key to export both type assertion functions and type test functions so we can support the full gamut of Typescript consumers. Type assertions are not supported <3.7.0 and are compiled away using `downlevel-dts`.